cloud/scope/clients.go

```diff
@@ -37,40 +38,68 @@ const (
 
 // AzureClients contains all the Azure clients used by the scopes.
 type AzureClients struct {
```

Comment:
I think this could use a rename and maybe some refactoring, the setCredentials func is a bit funky to me. out of scope for this pr

Reply:
yeah also SubscriptionID should follow the same pattern as `tenantID` etc., ie. 
```
func (c *AzureClients) SubscriptionID() string {
	return c.subscriptionID
}
```

and remove the SubscriptionID() func in cluster.go on scope

Reply:
I think the flip actually, sub isn't necessary for auth. remove it here and retrieve it from the cluster/control plane (clusterdescriber) spec

Reply:
(still not here though)

Reply:
oh nvm, I see what you mean